### PR TITLE
Switch to SCSS

### PR DESCRIPTION
**Pull request by matthewma7**
This resolve issue https://github.com/girder/girder_web_components/issues/41

I added linting ability as suggested by @zachmullen, but I failed to make the configuration part of the package.json like other options do. (Probably needs to modify the plugin  to support that)

### CODE REVIEW

**Comment by zachmullen** [comment]
Doesn't scss have an alternative syntax that lets us omit the curly braces? I'd like to use that if so, unless there's some reason the prefer this syntax.

**Comment by matthewma7** [comment]
The reasons I can think of are, one familiarity: I am more used to see scope with braces instead of with indentation. I feel this way it is natural when working alongside with javascript. and two interoperability: the SCSS syntax is a superset of CSS on syntax, so it is very easy for me to copy to/from browser dev tools in my daily development, and copy from some example code online sometimes. Overall, for my work habit, omitting colons and/or semicolons always meant more work for me.

**Comment by zachmullen** [comment]
Very good points. Combined with the fact that we are attempting to minimize the amount of CSS we write anyway, I think the syntax is less of a concern than other important factors, so I'm on board with this change.

**Comment by matthewma7** [comment]
Hi @zachmullen, If the PR looks good, should we approve it?

**Comment by jbeezley** [comment]
I don't feel strongly either way.  I'm in favor of switching to SCSS because it is more popular.  I mildly prefer the stylus-like syntax, but not enough to object.

File: .stylelintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+# .gitignore syntax (uses node-ignore behind the scenes)
```
Comment (zachmullen) [inline .stylelintignore]: Why is this file necessary?

File: package.json [hunk 2]
```diff
@@ -17,9 +18,10 @@
     "qs": "^6.5.2",
     "vue": "^2.5.16",
     "vue-async-computed": "^3.4.1",
-    "vuetify": "^1.1.9"
+    "vuetify": "^1.3.0"
```
Comment (zachmullen) [inline package.json]: This change doesn't have any effect, does it?

File: package.json [hunk 3]
```diff
@@ -29,11 +31,13 @@
...
+    "node-sass": "^4.9.0",
     "pug-lint-vue": "^0.1.3",
-    "stylus": "^0.54.5",
-    "stylus-loader": "^3.0.2",
+    "sass-loader": "^7.0.1",
+    "stylelint-config-standard": "^18.2.0",
     "vue-cli-plugin-pug": "^1.0.7",
-    "vue-template-compiler": "^2.5.16"
+    "vue-template-compiler": "^2.5.16",
+    "vuetify-scss": "^1.2.0"
```
Comment (zachmullen) [inline package.json]: Is this library necessary? We import the vuetify CSS already built, and we don't want to have to know about the upstream build toolchain unless we really have to.

File: vue.config.js [hunk 4]
```diff
@@ -1,3 +1,8 @@
 module.exports = {
   lintOnSave: false,
+
+  pluginOptions: {
+    lintStyleOnBuild: false,
+    stylelint: {},
```
Comment (zachmullen) [inline vue.config.js]: Aren't both of these the defaults? I'd vote to remove them if so.

File: stylelint.config.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  root: true,
+  extends: [
+    'stylelint-config-standard',
+  ],
+};
```
Comment (zachmullen) [inline stylelint.config.js]: It sounds like this could be added as a "stylelint" property under package.json, if so we should do that.

Comment (matthewma7) [reply stylelint.config.js]: Thanks, the key stylelint works.

File: vue.config.js (hunk 4 above)
Comment (matthewma7) [reply vue.config.js]: Yes, we can remove these. (But seems the pluginOptions needs to stay, or the @ascendancyy/vue-cli-plugin-stylelint will throw exception)

File: vue.config.js [hunk 6]
```diff
@@ -1,3 +1,6 @@
 module.exports = {
   lintOnSave: false,
+
+  pluginOptions: {
+  }
```
Comment (zachmullen) [inline vue.config.js]: We can just omit this block since it's now empty

File: package.json (hunk 2 above)
Comment (matthewma7) [reply package.json]: I think I accidentally upgraded Vuetify somehow, but I think we probably should update Vuetify in a probably PR for its fixes and new features.

File: vue.config.js (hunk 6 above)
Comment (matthewma7) [reply vue.config.js]: It looks like if I omit the key. The lint:style will throw an error `TypeError: Cannot destructure property `lintStyleOnBuild` of 'undefined' or 'null'.
    at module.exports (/home/matthew/girder_web_components/node_modules/@ascendancyy/vue-cli-plugin-stylelint/index.js:7:11)`

File: package.json (hunk 2 above)
Comment (zachmullen) [reply package.json]: Agreed, since the upgrade of Vuetify is out of scope of this PR, can you also back out those changes from the yarn.lock? We can add it on a second PR and then reviewers will know to poke around the components to make sure everything looks nice.

Comment (matthewma7) [reply package.json]: Great catch. I rebased again to try to clean up the yarn.lock.

File: package.json [hunk 7]
```diff
@@ -117,6 +120,12 @@
     "validateSelfClosingTags": true,
     "validateTemplateString": true
   },
+  "stylelint": {
+    "root": true,
```
Comment (zachmullen) [inline package.json]: The [documentation](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md) doesn't mention `root`, what does this do?

**Comment by zachmullen** [comment]
Just to try and distill down the arguments on this subject to make it easy for people to decide on this issue:

**Pros**
* Much more popular tool, more community support
* Support for stylesheet linting for our Vue components (currently impossible with Stylus)
* Plain-old-CSS syntax makes it easier to copy paste, and is more familiar for new contributors

**Cons**
* Requires a much larger, native build toolchain to be installed by npm. It's only a dev dependency, though, and it installs cleanly on both Debian and MacOS in my testing. The upside of that is that it ought to build our stylesheets faster, though that will likely be a negligible difference.
* No more shorthand syntax. To be fair, there were some uncommon cases in which the stylus syntax could cause an ambiguous parse, so at least we won't run into that.

File: package.json (hunk 7 above)
Comment (matthewma7) [reply package.json]: I personally didn't know. It was generated for me. The eslintConfig configuration section also has the root. Maybe it has something to do with vue-cli? I didn't find any document as well. https://github.com/vuejs/vue-cli/blob/1a8394434ab074d6d15412f17d8374e84cee9314/packages/%40vue/cli/lib/Generator.js#L99

Comment (zachmullen) [reply package.json]: I just read through the source of `@ascendancyy/vue-cli-plugin-stylelint`, and it seems pretty rough, and IMO unnecessary. It's a whole lot of code that does almost nothing beyond this:

`stylelint src/**/*.vue`

So my preference would be to drop `@ascendancyy/vue-cli-plugin-stylelint` and do

`"lint:style": "stylelint 'src/**/*.vue'"`

If we anticipate including CSS in files aside from just vue single file components, we can add more extensions to that glob expression.

Comment (zachmullen) [reply package.json]: If we cared about linting on build, I'd say it's worth using the plugin, but I'm not a fan of linters causing build errors.

Comment (matthewma7) [reply package.json]: Sounds good, I'll give it a try and see if everything works well without it.

Comment (matthewma7) [reply package.json]: Everything works great, include the autofix that I would use. I'll push the change.

Comment (matthewma7) [reply package.json]: Hi @zachmullen, this PR is ready for further review

**Comment by zachmullen** [comment]
I approve of these changes. I'd like to get @jbeezley @subdavis and @jeffbaumes to also approve it before we merge.

**Comment by jeffbaumes** [comment]
I approve.

**Comment by jbeezley** [comment]
LGTM

**Comment by matthewma7** [comment]
Thanks, Zach and everyone 👌

**Comment by zachmullen** [comment]
Thank you @matthewma7 for the change!
